packages/storefront/shared/src/helpers/composeMiddlewareUrl.ts

```diff
@@ -0,0 +1,49 @@
+import { CreateSdkOptions } from "../types";
+
+interface ComposeMiddlewareUrlParams {
+  options: CreateSdkOptions;
+  headers: Record<string, string | string[]>;
+}
+
+function calculateMiddlewareUrl({
```

Comment:
is this method a duplicate?

Reply:
that's a tough question :D this file is also present in `packages/storefront/packages/nuxt/src/runtime/utils/composeMiddlewareUrl.ts`. I didn't find a way to use there the method defined in the `shared`, because the whole file is then delivered within the module to the user, because its needed during the runtime. I thought for Nuxt that we may to generate a copy of it during the build time, but I've decided that it would be an overkill. Now we use it only in two places, I don't think that this file would often change in the future. However if you think that there is any better way to achieve it, I'm happy to change it